pkg/util/pod.go

```diff
@@ -0,0 +1,73 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"strings"
+)
+
+func IsPodReady(pod *corev1.Pod) bool {
+	conditionsTrue := 0
+	for _, cond := range pod.Status.Conditions {
+		if cond.Status == corev1.ConditionTrue && (cond.Type == corev1.ContainersReady || cond.Type == corev1.PodReady) {
+			conditionsTrue++
+		}
+	}
+	return conditionsTrue == 2
+}
+
+func IsPodError(pod *corev1.Pod) bool {
+	if pod.Status.Phase == corev1.PodFailed || pod.Status.Phase == corev1.PodUnknown {
+		return true
+	}
+	conditionsFalse := 0
+	for _, cond := range pod.Status.Conditions {
+		if pod.Status.Phase == corev1.PodRunning && cond.Status == corev1.ConditionFalse && (cond.Type == corev1.ContainersReady || cond.Type == corev1.PodReady) {
+			conditionsFalse++
+		}
+	}
+	return conditionsFalse == 2
+}
+
+func IsPodResourceError(pod *corev1.Pod) bool {
+	if pod.Status.Phase == corev1.PodFailed && strings.Contains(pod.Status.Reason, "OutOf") {
+		return true
+	}
+	for _, cond := range pod.Status.Conditions {
+		if cond.Status == corev1.ConditionFalse && cond.Type == corev1.PodScheduled && cond.Reason == corev1.PodReasonUnschedulable &&
+			(strings.Contains(cond.Message, "Insufficient cpu") || strings.Contains(cond.Message, "Insufficient memory")) {
+			return true
+		}
+	}
+	return false
+}
+
+func DeleteResourceOfPod(pod *corev1.Pod) {
+	for i := range pod.Spec.Containers {
+		pod.Spec.Containers[i].Resources.Requests = nil
+		pod.Spec.Containers[i].Resources.Limits = nil
```

Comment:
should we still keep the limit?

Reply:
We can't just keep limits, because k8s automatically assigns a memory request that matches the limit, when Container specifies its own resource limit, but does not specify a resource request. 

ref: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits